net/WebSocketHandler.hpp

```diff
@@ -393,10 +416,18 @@ class WebSocketHandler : public ProtocolHandlerInterface
                     if (_isClient)
                         LOG_WRN("Servers should not send pongs, only clients");
 
-                    _pingTimeUs = std::chrono::duration_cast<std::chrono::microseconds>
-                        (std::chrono::steady_clock::now() - _lastPingSentTime).count();
-                    LOGA_TRC(WebSocket, "Pong received: " << _pingTimeUs << " microseconds");
-                    gotPing(code, _pingTimeUs);
+                    const auto now = std::chrono::steady_clock::now();
+                    const int64_t us = std::chrono::duration_cast<std::chrono::microseconds>
+                                          (now - _lastPingSentTime).count();
+                    const double avg_us = _pingMicroS.add(_lastPingSentTime, static_cast<double>(us));
```

Comment:
That is a bug of mine, outch .. yes, should pass `static_cast<double>(us)/1000000.0`.  Great catch, thank you!
Note: My unit test didn't catch this either, b/c fluctuations are very low but I should have done a better job in interpretation of the values. So I take the blame here - bad mistake.

Otherwise .. , TimePoint misses denoting dimension of value 'v' in method 'add', yes. Re 'convoluted' .. that really depends whether this functionality is desired and so forth. std:chrono duration in seconds and double could/should  be used for clarity, yup.

Reply:
> That is a bug of mine, outch .. yes, should pass `static_cast<double>(us)/1000000.0`. Great catch, thank you! Note: My unit test didn't catch this either, b/c fluctuations are very low but I should have done a better job in interpretation of the values. So I take the blame here - bad mistake.

The `us` value in microseconds managed by TimePoint is correct! (and so is the unit test)

Sorry, your comment confused myself .. the value managed by TimePoint is of `arbitrary nature` (and dimension), as mentioned in `TimePoint::add` So here microseconds is used. 

The confusing part is probably that this value itself reflects time duration, however, `TimePoint::duration` is orthogonal, as it only refers over the measured total time and not the semantics of this value (ping response duration in microseconds).
So yes, annotations in TimePoint probably should be more emphasized, only the `add` method does. *sigh*